src/Searchlight/Attributes/SearchlightEngine.cs

```diff
@@ -44,6 +50,14 @@ public SearchlightEngine AddClass(Type type)
         public SyntaxTree Parse(FetchRequest request)
         {
             var source = FindTable(request.table);
+            if (request.pageSize == null)
+            {
+                request.pageSize = MaximumPageSize;
+            }
+            if (request.pageSize > MaximumPageSize)
+            {
+                throw new InvalidPageSize { PageSize = request.pageSize == null ? "not specified" : "larger than the allowed maximum pageSize, " + Convert.ToString(MaximumPageSize) };
```

Comment:
I don't think request.PageSize can ever be null with the if statement above.

Reply:
So with string concatentation the Convert.ToString isn't necessary...but I think a better way to write this would be to use the string interpolation https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

Reply:
Oh right, that makes sense, I see that now - I'll pull out that previous logic and update the Exception statement to use string interpolation (I like the interpolation approach too, thanks for sharing that documentation)